### PR TITLE
Fix sql

### LINKED ISSUE #298: Unexpected error when using GDAL 3.10.0rc1 [BUG]
With released GDAL 3.9.3, no error occurs in `oe_get_network`: 
[osmextract.Rcheck.GDAL393.zip](https://github.com/user-attachments/files/17586527/osmextract.Rcheck.GDAL393.zip)
However, when running the same CMD check with the same platform and the same `sf`, GDAL 3.10.0rc1 shows a failure with no `its_walking` features found: 
[osmextract.Rcheck.GDAL3100.zip](https://github.com/user-attachments/files/17586542/osmextract.Rcheck.GDAL3100.zip)
Please run your own checks with a GDAL 3.10.0 release candidate (currently rc2, probably the last) and see whether this is a problem. @agila5 @edzer

### PR DESCRIPTION
**Pull request by agila5**
Fix SQL syntax following the suggestions in #298 for GDAL 3.10. 

Fixes #298.

### CODE REVIEW

File: R/get-network.R [hunk 1]
```diff
@@ -144,6 +144,12 @@ oe_get_network = function(
 # accepted but have support among many mappers. More precise options are defined
 # by other tags. See also: https://wiki.openstreetmap.org/wiki/Bicycle#Bicycle_Restrictions
 
+# WARNING: Starting from GDAL 3.10, an expression like "foo NOT IN ('bar')"
+# evaluates as false, while previously it would evaluate as true. Therefore, in
```
Comment (Robinlovelace) [inline R/get-network.R:148]: Is there a link to the GDAL documentation that says something similar?

**Review by Robinlovelace — COMMENTED** [review]
Looks like progress, wonder if it passes..

File: R/get-network.R (hunk 1 above)
Comment (rouault) [reply R/get-network.R:148]: yes see https://github.com/OSGeo/gdal/blob/master/MIGRATION_GUIDE.TXT#L4

Comment (Robinlovelace) [reply R/get-network.R:148]: Great, let's add that link to the docs.

Comment (rouault) [reply R/get-network.R:148]: (suggestion: use a permalink or link to the tag: https://github.com/OSGeo/gdal/blob/v3.10.0RC3/MIGRATION_GUIDE.TXT)

Comment (agila5) [reply R/get-network.R:148]: Added permalink, thank you very much!

**Pull request merged**